### PR TITLE
feat!: v3

### PR DESCRIPTION
**Pull request by jacoobes**

### CODE REVIEW

**Comment by jacoobes** [comment]
honestly this pr is too big to review xD will rely on feedback of users in alpha

**Comment by jacoobes** [comment]
Has been tested extensively thanks to the community and all developers involved. Feel free to give it a light review if you're free

File: src/core/contracts/error-handling.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    /**
+     * Number of times the process should throw an error until crashing and exiting
+     */
+    keepAlive: number;
+
+    /**
+     * @deprecated 
+     * Version 4 will remove this method
+     */
+    crash(err :Error) : never
```
Comment (EvolutionX-10) [inline src/core/contracts/error-handling.ts:14]: did you run prettier?

File: src/core/contracts/module-manager.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { CommandMeta, CommandModule, CommandModuleDefs, Module } from '../../types/core-modules';
```
Comment (EvolutionX-10) [inline src/core/contracts/module-manager.ts:1]: shouldn't we import types like `import type`?

Comment (jacoobes) [reply src/core/contracts/module-manager.ts:1]: We should for cleanliness, not adding it doesnt seem to affect bundle output though

File: src/core/contracts/error-handling.ts (hunk 1 above)
Comment (jacoobes) [reply src/core/contracts/error-handling.ts:14]: nah, let me do it

**Review by EvolutionX-10 — APPROVED** [review]

**Pull request merged**